src/script_interface/virtual_sites/ActiveVirtualSitesHandle.hpp

```diff
@@ -0,0 +1,61 @@
+/*
+  Copyright (C) 2010,2011,2012,2013,2014 The ESPResSo project
+  Copyright (C) 2002,2003,2004,2005,2006,2007,2008,2009,2010
+  Max-Planck-Institute for Polymer Research, Theory Group
+
+  This file is part of ESPResSo.
+
+  ESPResSo is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  ESPResSo is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef SCRIPT_INTERFACE_VIRTUAL_SITES_ACTIVE_VIRTUAL_SITES_HANDLE_HPP
+#define SCRIPT_INTERFACE_VIRTUAL_SITES_ACTIVE_VIRTUAL_SITES_HANDLE_HPP
+
+#include "config.hpp"
+#include "auto_parameters/AutoParameters.hpp"
+#include "VirtualSites.hpp"
+#include "core/virtual_sites.hpp"
+#include "errorhandling.hpp"
+
+#ifdef VIRTUAL_SITES
+
+namespace ScriptInterface {
+namespace VirtualSites {
+
+class ActiveVirtualSitesHandle : public AutoParameters {
+public:
+  ActiveVirtualSitesHandle() {
+      add_parameters({
+          {"implementation", 
+                     [this](Variant const &value) {
+                       m_active_implementation =
```

Comment:
The type check is not needed as `get_value` will throw if value can not be cast into the correct type, you can just do `set_virtual_sites(get_value<std::shared_ptr<VirtualSites>>(value))`.